### PR TITLE
Add some cartography algorithms

### LINKED ISSUE #3566: [FEATURE][processing] Add new algorithm "Print layout map extent to layer"
Original commit: https://github.com/qgis/QGIS/commit/92c8fddac2bca80c60ff0978d34dc01e9db5bb79 by nyalldawson

This algorithm creates a polygon layer containing the extent
of a print layout map item, with attributes specifying the map
size (in layout units), scale and rotatation.

The main use case is when you want to create an advanced overview
indicator and the inbuilt layout tools to do this don't suffice.

### LINKED ISSUE #3964: [FEATURE][processing] New algorithm "Create style database from project"
Original commit: https://github.com/qgis/QGIS/commit/08a985ac8ffbc829abb57499687274f8649ae355 by nyalldawson

Extracts all symbols, color ramps, text formats and label settings from
the current project and stores them in a new style XML database

### LINKED ISSUE #3969: [processing][needs-docs] Allow running Style from Project algorithm
on a specific project file
Original commit: https://github.com/qgis/QGIS/commit/200e203496743a866d5f778d06305c6f1c1e066e by nyalldawson

Instead of forcing the algorithm to always use the current project,
add an option parameter allowing users to run the algorithm on a
specific project file instead.

### LINKED ISSUE #3971: [FEATURE][processing] New algorithm "Combine style databases"
Original commit: https://github.com/qgis/QGIS/commit/c828130f9ac161a0f5716043953a50c805523694 by nyalldawson

This algorithm combines multiple QGIS style databases into a
single style database. If any symbols exist with duplicate
names between the different source databases these will be
renamed to have unique names in the output combined database.

### PR DESCRIPTION
**Pull request by DelazJ**
Fixes #3971, fixes #3969, fixes #3964, fixes #3566

### CODE REVIEW

File: source/docs/user_manual/processing_algs/qgis/cartography.rst [hunk 1]
```diff
@@ -70,6 +70,120 @@ Outputs
   Lists symbols from the provided style database which could not match any category.
 
 
+.. _qgiscombinestyles:
+
+Combine style databasess
+---------------------------------------
+Combines multiple QGIS style databases into a single style database.
+If any items of the same type exist with duplicate names between the different source
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/cartography.rst]:
```suggestion
If items of the same type with the same name exist in different source
```

File: source/docs/user_manual/processing_algs/qgis/cartography.rst [hunk 2]
```diff
@@ -70,6 +70,120 @@ Outputs
...
+See also
+........
+:ref:`qgisstylefromproject`
+
+
+.. _qgisstylefromproject:
+
+Create style database from project
+-----------------------------------
+Extracts all style objects (including symbols, color ramps, text formats and
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/cartography.rst]:
```suggestion
Extracts all style objects (symbols, color ramps, text formats and
```

Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/cartography.rst]: Or are there more?

File: source/docs/user_manual/processing_algs/qgis/cartography.rst [hunk 3]
```diff
@@ -37,7 +37,7 @@ Parameters
   Field or expression to categorize the features.
 
 ``Style database (leave blank to use saved symbols)`` [file]
-  File (``.XML``) containing the symbols to apply to the input layer categories.
+  File (:file`.XML`) containing the symbols to apply to the input layer categories.
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/cartography.rst]:
```suggestion
  File (:file:`XML`) containing the symbols to apply to the input layer categories.
```

File: source/docs/user_manual/processing_algs/qgis/cartography.rst [hunk 4]
```diff
@@ -70,6 +70,120 @@ Outputs
...
+........
+:ref:`qgiscombinestyles`
+
+
+.. _qgisprintlayoutmapextenttolayer:
+
+Print layout map extent to layer
+--------------------------------
+
+Creates a polygon layer containing the extent of a print layout map item
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/cartography.rst]:
```suggestion
Creates a polygon layer containing the bounding box of a print layout map item
```

Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/cartography.rst]: ?

File: source/docs/user_manual/processing_algs/qgis/cartography.rst [hunk 5]
```diff
@@ -70,6 +70,120 @@ Outputs
...
+
+  The map item(s) whose information you want to extract.
+  If none is provided then all the map items are processed.
+
+  Default: *All the map items*
+
+``Override CRS`` [crs]
+  Optional
+
+  Select the CRS in which the information will be reported.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/cartography.rst]:
```suggestion
  Select the CRS of the layer in which the information will be reported.
```

File: source/docs/user_manual/processing_algs/qgis/cartography.rst (hunk 4 above)
Comment (DelazJ) [reply source/docs/user_manual/processing_algs/qgis/cartography.rst]: Isn't the bounding box of a map item same as its extent?

File: source/docs/user_manual/processing_algs/qgis/cartography.rst (hunk 2 above)
Comment (DelazJ) [reply source/docs/user_manual/processing_algs/qgis/cartography.rst]: No afaik. They were two four months ago but not sure there are more currently.

File: source/docs/user_manual/processing_algs/qgis/cartography.rst (hunk 4 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/cartography.rst]: "Extent" can also be used for a polygon. "The geographical extent of India".  So I would say "bounding box" is more precise.

**Pull request merged**